main.js

```diff
@@ -9,73 +13,93 @@ const userInfo = {
 const loginForm = document.querySelector("#login")
 const usernameInput = document.querySelector("#username");
 const userPassword = document.querySelector("#password");
+const submitButton = document.querySelector("#login_btn");
+
+document.addEventListener("DOMContentLoaded", function(){
+    usernameInput.value = localStorage.getItem("userInput");
+})
+
+loginForm.addEventListener("click", function(e) {
+    if(e.target === usernameInput || e.target === userPassword || e.target === submitButton){
+        touchedElements.push(e.target);
+    }
+    if(e.target === submitButton){
+        removeError();
+    }
+});
 
 loginForm.addEventListener("submit",function(e){
-    
-     e.preventDefault();
+    e.preventDefault();
+
+    formValues.valueUsername = usernameInput.value;
+    formValues.valuePassword = userPassword.value;
+    localStorage.setItem("userInput", formValues.valueUsername);
 
-    if (isValidUsername() && isValidPassword()) {
-        if (checkData()) {
+    if (isValidUsername() & isValidPassword()) {
+        if (checkUsername() & checkPassword()) {
             userInfo.token = "xxx"
             alert("correct");
-        } else{
-            fieldsClear();
         }
-    } else{
-        fieldsClear();
-    }    
-    
+    }  
 });
 
 function isValidUsername() {
     const nameRegex = /^[a-zA-Z\-]+$/;
 
     if(!usernameInput.value){
-        showError("form_username","The username is empty");
+        validationErrors.errorUsername = "The username is empty";
```

Comment:
Let's create special function for this action.

`function setValidationError(fieldName, error)`

Reply:
I want you to make your form completely dependent on these variables (`validationErrors`, `formValues`, `touchedElements`). So it will look quite dangerous when you change dependencies directly. `setValidationError` will provide a special interface with which you can manipulate your dependencies.

It would be much safe for naming change like `validationErrors` -> `shitShitValidationErrors` or change js variable to localStorage. For this you need to change only the stuff of the `setValidationError`

Reply:
Done